src/Common/Ian/Ian_Client.php

```diff
@@ -0,0 +1,386 @@
+			// }
+
+			// $notifications = Conditionals::filter_ian_feed( $body['notifications'] );
+
+			// $this->slug = $slug;
+			// $this->has_user_dismissed( get_current_user_id() )
+
+			// If we want to reset all dismissible content for testing.
+			// delete_user_meta( get_current_user_id(), 'tec-dismissible-content' );
+			// or just one
+			// $this->undismiss( get_current_user_id() )
+
+			// TODO: Below is an example notifications array. Send the real one.
+			wp_send_json_success(
+				[
+					[
+						'id'          => '101',
+						'type'        => 'update',
+						'slug'        => 'tec-update-664',
+						'title'       => 'The Events Calendar 6.6.4 Update',
+						'content'     => '<p>The latest update of The Events Calendar adds an option to allow for duplicate Venue creation, updates custom table query logic to avoid DB error, and logic that displays the “REST API blocked” banner to prevent false positives.</p>',
+						'actions'     => [
+							[
+								'text'   => 'See Details',
+								'link'   => 'https://evnt.is/1ai-',
+								'target' => '_blank',
+							],
+							[
+								'text'   => 'Update Now',
+								'link'   => '/wp-admin/update-core.php',
+								'target' => '_self',
+							],
+						],
+						'dismissible' => true,
+					],
+					[
+						'id'          => '102',
+						'type'        => 'notice',
+						'slug'        => 'event-tickets-upsell',
+						'title'       => 'Sell Tickets & Collect RSVPs with Event Tickets',
+						'content'     => '<p>Sell tickets, collect RSVPs and manage attendees for free.</p>',
+						'actions'     => [
+							[
+								'text'   => 'Learn More',
+								'link'   => 'https://evnt.is/1aj1',
+								'target' => '_blank',
+							],
+						],
+						'dismissible' => true,
+					],
+					[
+						'id'          => '103',
+						'type'        => 'warning',
+						'slug'        => 'fbar-upgrade-556',
+						'title'       => 'Filter Bar 5.5.6 Security Update',
+						'content'     => '<p>Get the latest version of Filter Bar for important security updates.</p>',
+						'actions'     => [
+							[
+								'text'   => 'Update',
+								'link'   => '/wp-admin/plugins.php?plugin_status=upgrade',
+								'target' => '_self',
+							],
+						],
+						'dismissible' => false,
+					],
+				],
+				200
+			);
+		} else {
+			wp_send_json_error( 'Invalid nonce', 403 );
+		}
+	}
+
+	/**
+	 * AJAX handler for dismissing IAN notifications.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function handle_dismiss(): void {
+		$nonce = isset( $_POST['nonce'] ) ? sanitize_text_field( wp_unslash( $_POST['nonce'] ) ) : '';
+
+		if ( wp_verify_nonce( $nonce, 'common_ian_nonce' ) ) {
+
+			$slug = isset( $_POST['slug'] ) ? sanitize_key( $_POST['slug'] ) : '';
+
+			if ( empty( $slug ) ) {
+				wp_send_json_error( 'Invalid slug', 403 );
+			}
+
+			$this->slug = $slug;
+
+			wp_send_json_success( $this->dismiss(), 200 );
+		} else {
+			wp_send_json_error( 'Invalid nonce', 403 );
+		}
+	}
```

Comment:
We need to have one nonce per notification, attach the nonce to the dimiss button itself.